packages/core/src/tools/ToolsManager.ts

```diff
@@ -1,42 +1,223 @@
-import type { BlockToolConstructor, InlineToolsConfig } from '@editorjs/sdk';
+import 'reflect-metadata';
+import { deepMerge, isFunction, isObject, PromiseQueue } from '@editorjs/helpers';
+import { Inject, Service } from 'typedi';
+import {
+  BlockToolFacade, BlockTuneFacade,
+  InlineToolFacade,
+  ToolsCollection,
+  ToolsFactory
+} from './facades/index.js';
 import { Paragraph } from './internal/block-tools/paragraph/index.js';
-import type { EditorConfig } from '@editorjs/editorjs';
+import type {
+  EditorConfig,
+  ToolConstructable,
+  ToolSettings
+} from '@editorjs/editorjs';
 import BoldInlineTool from './internal/inline-tools/bold/index.js';
 import ItalicInlineTool from './internal/inline-tools/italic/index.js';
+import { BlockToolConstructor, InlineTool, InlineToolConstructor } from '@editorjs/sdk';
+import { UnifiedToolConfig } from '../entities/index.js';
 
 /**
  * Works with tools
+ * @todo - validate tools configurations
+ * @todo - merge internal tools
  */
+@Service()
 export default class ToolsManager {
-  #tools: EditorConfig['tools'];
+  #factory: ToolsFactory;
+  #config: UnifiedToolConfig;
+  #availableTools = new ToolsCollection();
+  #unavailableTools = new ToolsCollection();
 
   /**
-   * @param tools - Tools configuration passed by user
+   * Returns available Tools
    */
-  constructor(tools: EditorConfig['tools']) {
-    this.#tools = tools;
+  public get available(): ToolsCollection {
+    return this.#availableTools;
   }
 
   /**
-   * Returns a block tool by its name
-   * @param toolName - name of a tool to resolve
+   * Returns unavailable Tools
    */
-  public resolveBlockTool(toolName: string): BlockToolConstructor {
-    switch (toolName) {
-      case 'paragraph':
-        return Paragraph;
-      default:
-        throw new Error(`Unknown tool: ${toolName}`);
+  public get unavailable(): ToolsCollection {
+    return this.#unavailableTools;
+  }
+
+  /**
+   * Return Tools for the Inline Toolbar
+   */
+  public get inlineTools(): ToolsCollection<InlineToolFacade> {
+    return this.available.inlineTools;
+  }
+
+  /**
+   * Return editor block tools
+   */
+  public get blockTools(): ToolsCollection<BlockToolFacade> {
+    return this.available.blockTools;
+  }
+
+  /**
+   * Return available Block Tunes
+   * @returns - object of Inline Tool's classes
+   */
+  public get blockTunes(): ToolsCollection<BlockTuneFacade> {
+    return this.available.blockTunes;
+  }
+
+  /**
+   * Returns internal tools
+   */
+  public get internal(): ToolsCollection {
+    return this.available.internalTools;
+  }
+
+  /**
+   * @param editorConfig - EditorConfig object
+   * @param editorConfig.tools - Tools configuration passed by user
+   */
+  constructor(@Inject('EditorConfig') editorConfig: EditorConfig) {
+    this.#config = this.#prepareConfig(editorConfig.tools ?? {});
+
+    this.#validateTools();
+
+    this.#factory = new ToolsFactory(this.#config, editorConfig, {});
+
+    void this.#prepareTools();
```

Comment:
im not sure that this promise should be ignored. Maybe you can await it in somewhere Core

Reply:
I made the method public, let's await it once `isReady` api is implemented